library/network_connections.py

```diff
@@ -2015,21 +2042,47 @@ def _check_ethtool_setting_support(self, idx, connection):
         if "ethtool" not in connection:
             return
 
-        ethtool_features = connection["ethtool"]["features"]
-        specified_features = dict(
-            [(k, v) for k, v in ethtool_features.items() if v is not None]
-        )
+        ethtool_dict = {
+            "features": nm_provider.get_nm_ethtool_feature,
+            "set-ring": nm_provider.get_nm_ethtool_setring
+        } 
+
+        for ethtool_key, nm_get_name_fcnt in ethtool_dict.items():
+            ethtool_settings = connection["ethtool"][ethtool_key]
+            specified = dict(
+                [(k, v) for k, v in ethtool_settings.items() if v is not None]
+            )
 
-        if specified_features and not hasattr(NM, "SettingEthtool"):
-            self.log_fatal(idx, "ethtool.features specified but not supported by NM")
 
-        for feature, setting in specified_features.items():
-            nm_feature = nm_provider.get_nm_ethtool_feature(feature)
-            if not nm_feature:
+            if specified and not hasattr(NM, "SettingEthtool"):
                 self.log_fatal(
-                    idx, "ethtool feature %s specified but not support by NM" % feature
+                    idx, "ethtool.%s specified but not supported by NM", specified
                 )
 
+            for option, _ in specified.items():
+                nm_name = nm_get_name_fcnt(option)
+                if not nm_name:
+                    self.log_fatal(
+                        idx,
+                        "ethtool %s setting %s specified "
+                        "but not supported by NM" % (ethtool_key, option),
+                    )
+
+        # ethtool_features = connection["ethtool"]["features"]
+        # specified_features = dict(
+        #     [(k, v) for k, v in ethtool_features.items() if v is not None]
+        # )
+
+        # if specified_features and not hasattr(NM, "SettingEthtool"):
+        #     self.log_fatal(idx, "ethtool.features specified but not supported by NM")
+
+        # for feature, setting in specified_features.items():
+        #     nm_feature = nm_provider.get_nm_ethtool_feature(feature)
+        #     if not nm_feature:
+        #         self.log_fatal(
+        #             idx, "ethtool feature %s specified but not support by NM" % feature
+        #         )
+
```

Comment:
Why is this code commented? If it is not needed, please remove it.

Reply:
The commented code was removed.